stonesoup/predictor/particle.py

```diff
@@ -91,3 +97,137 @@ def predict(self, prior, *args, **kwargs):
                                      timestamp=particle_prediction.timestamp,
                                      fixed_covar=kalman_prediction.covar, particle_list=None,
                                      transition_model=self.transition_model)
+
+
+class MultiModelPredictor(Predictor):
+    """MultiModelPredictor class
+
+    An implementation of a Particle Filter predictor.
+    """
+    transition_model = None
+    transition_models: Sequence[TransitionModel] = Property(
+        doc="Transition models used to for particle transition, selected by model index on "
+            "particle."
+    )
+    transition_matrix: np.ndarray = Property(
+        doc="n-model by n-model transition matrix."
+    )
+    position_mappings: Sequence[Sequence[int]] = Property(
+        doc="Sequence of position mappings associated with each transition model.")
```

Comment:
The term "position_mappings" causes a bit of confusion with me.  For example, if you have a transition model for a 4dim state vector (x, x_v, y, y_v) such as _CombinedLinearGaussianTransitionModel(ConstantVelocity(0.05), ConstantVelocity(0.05))_ you would want your position_mapping to be [0,1,2,3], not an intuitive [0, 2]. Maybe a change of name will make it more obvious? model_mappings?

Reply:
Hopefully improved this in 67ae107